timon/plugins/base.py

```diff
@@ -47,3 +49,22 @@ async def stop_plugin(self):
     async def stop(self):
         raise NotImplementedError(
             "Plugin %s doesn't have a start method implemented", self.name)
+
+
+class OnDbStorePlugin:
```

Comment:
Any reason this is a mixin and not inherited from TimonBasePlugin?

Reply:
it could, I wasn't sure; but it's a mixin so by definition it would not inherit from `TimonBasePlugin`

say at some point we add an other capability, it would be in an other mixin like `OnBidoofPlugin`; if both `OnDbStorePlugin` and `OnBidoofPlugin` are implemented by a user plugin then you would have an inheritance "diamond of death" situation; that's why mixin does not inherit from a base class

Reply:
if it's a mixin, then  let's name it like a mixin. 
If not one might think, that `OnDbStorePlugin` is a plugin that stores in a BD.

It seems an often used convention is to end class names with `MixIn` if they are a mix in.

https://chatgpt.com/share/67162299-04c8-800c-a7de-03b07a743af8